lib/govuk_schemas/random_schema_generator.rb

```diff
@@ -116,12 +116,27 @@ def generate_random_object(subschema)
     def generate_random_array(props)
       min = props["minItems"] || 0
       max = props["maxItems"] || 10
+      unique = props["uniqueItems"] == true
       num_items = @random.rand(min..max)
+      items = []
+      attempts = 0
+      max_attempts = num_items * 100
 
-      num_items.times.map do
+      until items.length == num_items
         # sometimes arrays don't have `items` specified, not sure if this is a bug
-        generate_value(props["items"] || {})
+        new_value = generate_value(props["items"] || {})
+
+        if unique && items.include?(new_value)
+          attempts += 1
+          raise "Failed to create a unique array item after #{max_attempts} attempts" if attempts >= max_attempts
```

Comment:
Is it possible to avoid the need to raise here I wonder.  I'm sure I'm missing something, but just thinking if it's possible to do something like this...

```
min = props["minItems"] || 0
max = props["items"].uniq.count
unique = props["uniqueItems"] == true
num_items = @random.rand(min..max)
generate_value(props["items"].uniq.sample(num_items))
```

Reply:
I think you perhaps have a misinterpretation of what `props["items"]` contains. I'm getting the impression you're expecting it to contain an array of possible items.

Whereas it actually contains a schema definition. The most common example is probably `{ "type" => "string" }`. So there are a few possible scenarios where you could generate a schema that causes an eternal loop.

Reply:
Ah - got ya!  Thanks for clarifying.